Program.cs

```diff
@@ -325,10 +325,29 @@ static void Setup(string username = "")
 			var password = Console.ReadLine();
 
 			UserLogin login = new UserLogin(username, password);
-			Console.Write($"Logging in {username}... ");
-			LoginResult loginResult = login.DoLogin();
-			Console.WriteLine(loginResult);
-			if (!login.LoggedIn) return;
+			string emailCode = null, twoFactorCode = null;
+			while (true)
+			{
+				login.EmailCode = emailCode;
+				login.TwoFactorCode = twoFactorCode;
+				Console.Write($"Logging in {username}... ");
+				LoginResult loginResult = login.DoLogin();
+				Console.WriteLine(loginResult);
+				if (loginResult == LoginResult.NeedEmail)
+				{
+					Console.Write("Email code: ");
+					emailCode = Console.ReadLine();
+					continue;
+				}
+				if (loginResult == LoginResult.Need2FA)
+				{
+					Console.Write("2FA code: ");
+					twoFactorCode = Console.ReadLine();
+					continue;
+				}
+				if (!login.LoggedIn) return;
+				break;
```

Comment:
No, we use `continue` to retry, and `break` to stop the loop once we can proceed. This is often used as a replacement for `goto`.